ufl/argument.py

```diff
@@ -54,81 +59,63 @@ def __init__(self, function_space, number, part=None):
         self._ufl_shape = function_space.ufl_element().value_shape()
 
         if not isinstance(number, numbers.Integral):
-            error("Expecting an int for number, not %s" % (number,))
+            error(f"Expecting an int for number, not {number}")
         if part is not None and not isinstance(part, numbers.Integral):
-            error("Expecting None or an int for part, not %s" % (part,))
+            error(f"Expecting None or an int for part, not {part}")
         self._number = number
         self._part = part
 
-        self._repr = "Argument(%s, %s, %s)" % (
+        self._repr = "Argument({0:s}, {1:s}, {2:s})".format(
             repr(self._ufl_function_space), repr(self._number), repr(self._part))
 
     @property
     def ufl_shape(self):
-        "Return the associated UFL shape."
+        """Return the associated UFL shape."""
         return self._ufl_shape
 
     def ufl_function_space(self):
-        "Get the function space of this Argument."
+        """Get the function space of this Argument."""
         return self._ufl_function_space
 
-    def ufl_domain(self):
-        "Deprecated, please use .ufl_function_space().ufl_domain() instead."
-        # TODO: deprecate("Argument.ufl_domain() is deprecated, please
-        # use .ufl_function_space().ufl_domain() instead.")
-        return self._ufl_function_space.ufl_domain()
-
-    def ufl_element(self):
-        "Deprecated, please use .ufl_function_space().ufl_element() instead."
-        # TODO: deprecate("Argument.ufl_domain() is deprecated, please
-        # use .ufl_function_space().ufl_element() instead.")
```

Comment:
TBH, I think this deprecation comment seems misguided. I think that there should be a consistent way to get the domain from any expression, and it should probably be `ufl_domains`. Otherwise you have to special case arguments everywhere.

Reply:
what was the resolution to this?